src/treetensornetworks/solvers/tdvp.jl

```diff
@@ -118,11 +112,33 @@ function tdvp(
   nsite=2,
   nsteps=nothing,
   order::Integer=2,
+  (sweep_observer!)=observer(),
   kwargs...,
 )
-  nsweeps = _compute_nsweeps(nsteps, t, time_step)
+  nsweeps = _compute_nsweeps(nsteps, t, time_step, order)
   sweep_regions = tdvp_sweep(order, nsite, time_step, init; kwargs...)
-  return alternating_update(solver, H, init; nsweeps, sweep_regions, nsite, kwargs...)
+
+  function sweep_time_printer(; outputlevel, sweep, kwargs...)
+    if outputlevel >= 1
+      sweeps_per_step = order ÷ 2
+      if sweep % sweeps_per_step == 0
+        current_time = (sweep / sweeps_per_step) * time_step
+        println("Current time (sweep $sweep) = ", round(current_time; digits=3))
+      end
+    end
+    return nothing
+  end
+
+  insert_function!(sweep_observer!, "sweep_time_printer" => sweep_time_printer)
+
+  psi = alternating_update(
+    solver, H, init; nsweeps, sweep_observer!, sweep_regions, nsite, kwargs...
+  )
+
+  # remove _current_time_printer
```

Comment:
I guess that should be `sweep_time_printer`?

Reply:
Good catch, fixed!